src/REL/d/a/obj/d_a_obj_block_underground.cpp

```diff
@@ -1,6 +1,292 @@
+    if (plc == nullptr) {
+        return false;
+    }
+
+    bool result = !mBgW.Set((cBgD_t *)dzb, (PLC *)plc, cBgW::MOVE_BG_e, &mWorldMtx, &mScale);
+    if (!result) {
+        return false;
+    }
+
+    if (fn_459_A00()) {
+        dCcD_Cyl *cyl = new dCcD_Cyl();
+        mCyl = cyl;
+        if (mCyl == nullptr) {
+            return false;
+        }
+    }
+    return true;
+}
+
+int dAcOBlockUnderground::actorCreate() {
+    u8 sceneFlag = getSecondSceneFlag();
+    if (sceneFlag < 0xff) {
+        bool flag = SceneflagManager::sInstance->checkFlag(roomid, sceneFlag);
+        if (flag) {
+            return FAILED;
+        }
+    }
+
+    bool success = false;
+    s32 newSubtype = 0;
+    u8 currentSubtype = getSubtype();
+    const s32 *subtype = &sSubtypeIdx[0];
+    while ((!success && newSubtype <= 3)) {
+        if (currentSubtype == *subtype) {
+            success = true;
+        } else {
+            subtype += 3;
+            newSubtype++;
+        }
+    }
+    if (success) {
+        setSubtype(newSubtype);
+    } else {
+        return FAILED;
+    }
+
+    updateMatrix();
+    CREATE_ALLOCATOR(dAcOBlockUnderground);
+    mMdl.setLocalMtx(mWorldMtx);
+    mMdl.setPriorityDraw(0x82, 0x7f);
+    if (fn_459_A00()) {
+        mStts.SetRank(0xD);
+        mCyl->Set(sCylSrc);
+        mCyl->SetStts(mStts);
+        if (fn_459_9A0()) {
+            mCyl->mTg.mSrc.mType = getCylTgType();
+        }
+        mCyl->SetC(position);
+    }
+    mBgW.Move();
+    mVec3_c min, max;
+    mMdl.getBounds(&min, &max);
+    boundingBox.Set(min, max);
+    return SUCCEEDED;
+}
+
+int dAcOBlockUnderground::actorPostCreate() {
+    if (mBgW.ChkUsed()) {
+        mBgW.Lock();
+        return SUCCEEDED;
+    }
+    registBg();
+    return NOT_READY;
+}
+
+int dAcOBlockUnderground::actorExecute() {
+    if (fn_459_A00()) {
+        if (mCyl->ChkTgHit()) {
+            if (mUndergroundRef.p_owner != nullptr) {
+                mUndergroundRef.get()->fn_458_9D0(mField_0x57C, mField_0x57E);
+            }
+
+            if (getSubtype() == 1) {
+                spawnEffect(PARTICLE_RESOURCE_ID_MAPPING_461_);
+                dSndSmallEffectMgr_c::GetInstance()->playSoundAtPosition(0xaa6, position);
+            } else {
+                spawnEffect(PARTICLE_RESOURCE_ID_MAPPING_462_);
+                dSndSmallEffectMgr_c::GetInstance()->playSoundAtPosition(0xaa7, position);
+            }
+
+            u8 firstSceneFlag = getFirstSceneFlag();
+            if (firstSceneFlag < 0xFF) {
+                SceneflagManager::sInstance->setFlag(roomid, firstSceneFlag);
+            }
+
+            u8 secondSceneFlag = getSecondSceneFlag();
+            if (secondSceneFlag < 0xFF) {
+                SceneflagManager::sInstance->setFlag(roomid, secondSceneFlag);
+            }
+
+            deleteRequest();
+        } else if (dAcPy_c::GetLink()->checkActionFlagsCont(dAcPy_c::FLG0_SWING_ROPE)) {
```

Comment:
```suggestion
        } else if (dAcPy_c::GetLink()->checkActionFlagsCont(0x400000)) {
```
we don't have a flag enum for the `*Cont` flags yet.